src/LuaRenderer/nodes/statements/renderFunctionDeclaration.ts

```diff
@@ -1,16 +1,17 @@
 import * as lua from "LuaAST";
+import { assert } from "Shared/util/assert";
 import { render, RenderState } from "LuaRenderer";
 import { renderParameters } from "LuaRenderer/util/renderParameters";
 import { renderStatements } from "LuaRenderer/util/renderStatements";
 
 export function renderFunctionDeclaration(state: RenderState, node: lua.FunctionDeclaration) {
-	const hasLocal = node.localize && lua.isAnyIdentifier(node.name);
+	const hasLocal = node.localize && assert(lua.isAnyIdentifier(node.name), "local function cannot be a property");
```

Comment:
Hmm I'm realizing now that `node.localize` probably should've been like `node.tryLocalize`?
Or maybe flip the condition so that it's `doNotLocalize`.
The point was that under certain conditions we never want to add `local`. But `localize: true` didn't necessarily mean we _have_ to add `local`.

Right now, we aren't using non-identifiers for function names, so this doesn't really matter?
I'd say let's keep the assert for now, I just wanted to discuss my original reasoning.